tools/compression_advisor.py

```
@@ -0,0 +1,1012 @@
+
+    For dict-aware compressors, looks for the ``past`` dict entry first
+    (= the currently deployed dictionary) and falls back to ``none`` if
+    no ``past`` match exists. For plain compressors, uses the ``none``
+    bucket only.
+
+    Returns 0.0 if the table is compressed but no matching entry is found.
+    """
+    if current_config.get("uncompressed"):
+        return 1.0
+
+    def _match(r, dict_mode):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_match`

(ANN202)